tests/models/layers/test_flash_attn.py

```diff
@@ -253,3 +254,107 @@ def test_sliding_window(sliding_window_size: int):
                      ) <= 1e-2 + 1e-2 * torch.norm(key_2.grad)
     assert torch.norm(value_2.grad - value_1.grad  # type: ignore
                      ) <= 1e-2 + 1e-2 * torch.norm(value_2.grad)
+
+
+@pytest.mark.gpu
+@pytest.mark.skipif(
+    not is_flash_v2_installed(v2_version='v2.4.2'),
+    reason='ALiBi only supported by Flash Attention after v2.4.2.')
+@pytest.mark.parametrize('n_heads', [1, 6, 8])
+def test_alibi_bias(n_heads: int):
+    # Test that sliding window attention works as expected.
+    dtype = torch.bfloat16
+    device = 'cuda'
+    d = 128
+    seqlen_1 = 8
+    bsz = 2
+
+    query_1 = torch.randn(bsz, seqlen_1, n_heads * d).to(dtype=dtype,
+                                                         device=device)
+    query_1.requires_grad = True
+    key_1 = torch.randn(bsz, seqlen_1, n_heads * d).to(dtype=dtype,
+                                                       device=device)
+    key_1.requires_grad = True
+    value_1 = torch.randn(bsz, seqlen_1, n_heads * d).to(dtype=dtype,
+                                                         device=device)
+    value_1.requires_grad = True
+    attn_bias_1 = gen_slopes(n_heads=n_heads,
+                             alibi_bias_max=8,
+                             device=torch.device(device),
+                             return_1d=True)
+    output_1, _, _ = flash_attn_fn(query=query_1,
+                                   key=key_1,
+                                   value=value_1,
+                                   n_heads=n_heads,
+                                   kv_n_heads=n_heads,
+                                   past_key_value=None,
+                                   softmax_scale=1 / math.sqrt(d),
+                                   attn_bias=attn_bias_1,
+                                   key_padding_mask=None,
+                                   is_causal=True,
+                                   dropout_p=0.0,
+                                   training=False,
+                                   needs_weights=False,
+                                   multiquery=False,
+                                   attention_mask_in_length=None,
+                                   should_repeat_kv_for_gqa=True)
+
+    output_1.sum().backward()
+
+    query_2 = query_1.detach().clone()
+    query_2.requires_grad = True
+    key_2 = key_1.detach().clone()
+    key_2.requires_grad = True
+    value_2 = value_1.detach().clone()
+    value_2.requires_grad = True
+
+    def gen_bias():
+        causal = True
+        bs = attention.attn_bias_shape('triton',
```

Comment:
add attn_bias_shape to the import above.
```suggestion
        bs = attn_bias_shape('triton',
```